pkg/member/member_add.go

```diff
@@ -0,0 +1,158 @@
+package member
+
+import (
+	"context"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/gardener/etcd-backup-restore/pkg/etcdutil"
+	brtypes "github.com/gardener/etcd-backup-restore/pkg/types"
+	"github.com/sirupsen/logrus"
+	"go.etcd.io/etcd/etcdserver/api/v3rpc/rpctypes"
+)
+
+const (
+	// RetryPeriod is the peroid after which an operation is retried
+	RetryPeriod time.Duration = 5 * time.Second
+)
+
+// AddMemberAsLearner add a member as a learner to the etcd cluster
+func AddMemberAsLearner(logger *logrus.Logger) error {
+	//Add member as learner to cluster
+	memberURL := getMemberURL()
+	if memberURL == "" {
+		logger.Warn("Could not fetch member URL")
+	}
+	for {
+		//Create etcd client
+		//TODO: Use secure transport
+		clientFactory := etcdutil.NewFactory(brtypes.EtcdConnectionConfig{
```

Comment:
Hi @unmarshall ,
Basically `NewFactory()` takes the `EtcdConnectionConfig` and returns the `clientFactory`. The thing is `EtcdConnectionConfig` can vary according to the use case. so, If we keep using the same factory this might leads to ambiguity and can leads to human error that dev don’t notice about `EtcdConfiguration` used in factory creation and ends up using same factory again and again.  I think with increasing code base it is better to create clientFactory again with `EtcdConnectionConfig`.